addons/account/models/account_move.py

```diff
@@ -699,7 +699,23 @@ def _get_quick_edit_suggestions(self, amount_total, quick_edit_total_amount, mov
                 taxes = journal.company_id.account_sale_tax_id if journal.type == 'sale' else journal.company_id.account_purchase_tax_id
             taxes = self.env['account.fiscal.position'].browse(fiscal_position_id).map_tax(taxes)
         price_untaxed = taxes.with_context(force_price_include=True).compute_all(quick_edit_total_amount - amount_total)['total_excluded']
-        return {'account_id': account_id, 'tax_ids': taxes.ids, 'price_unit': price_untaxed}
+
+        invoice_date = date.today()
+        prev_move = self.search([('state', '=', 'posted'),
+                                 ('journal_id', '=', journal_id),
+                                 ('move_type', '=', move_type),
+                                 ('company_id', '=', journal.company_id.id),
+                                 ('invoice_date', '!=', False)],
+                                order='date desc, name desc', limit=1)
+        if prev_move:
+            lock_date = prev_move.company_id.fiscalyear_lock_date
+            prev_date_last_day_of_month = prev_move.invoice_date + relativedelta(day=31)
+            if lock_date and prev_date_last_day_of_month <= lock_date:
+                invoice_date = lock_date + relativedelta(months=1) + relativedelta(day=31)
+            elif prev_date_last_day_of_month < date.today():
+                invoice_date = prev_date_last_day_of_month
```

Comment:
Have a look at `_get_accounting_date`
* you are not taking into account all the lock dates
* you are not taking into account the format of the sequence (which differs by default between client/vendor)

Reply:
Unfortunately, at this point many fields are `False` so calling this function fails. Furthermore, in our case, the sequence will be updated according to the suggested date, not the other way around

Reply:
However I'm changed the lock date to use `_get_user_fiscal_lock_date` to take into account the different lock dates possible in a `company`